src/common/image.cpp

```diff
@@ -152,13 +155,74 @@ bool wxImage::Create(const char* const* xpmData)
 #endif
 }
 
+// Check if multiplying two int values would overflow.
+// Return true if a*b overflows, false otherwise.
```

Comment:
This could probably be indeed useful elsewhere too, so perhaps we should move it to `wx/math.h`? We could probably keep it inline for simplicity (we don't have any `math.cpp`...) and because it could help the compiler to optimize it better (even without LTO).

Anyhow, please let me know if you'd like to extract this function (in which case we'd need to add documentation and, ideally, some tests, for it too) to the header in this PR or maybe do it in a later one.

Reply:
I think having a` wx/math.h` or `wx/safemath.h` would be good, but adding a new public API is beyond my abilities right now.  I also don't know enough about safe math to design a good public API...I'd probably just keep copying from [INT32-C](https://wiki.sei.cmu.edu/confluence/display/c/INT32-C.+Ensure+that+operations+on+signed+integers+do+not+result+in+overflow).  I also don't know how to implement wxWidgets code to take advantage of compiler-specific or target architecture capabilities (for example, using gcc's `__builtin_smul_overflow` or using an integer type that's twice as big at `int` if one is available).  In short, if it's going to be me, it shouldn't be now.  And I'd feel better starting with an internal API so that it's easier to change if I get it wrong.

On the other hand, I do think I should update the PR to remove the trailing whitespace that I accidently added, as well as add back the "/" that I had accidently had removed.  I'll have to figure out how to do that.

Reply:
OK, thanks David, I'll merge it as is then and will fix the spaces etc when I do it.

FWIW fixing this is really simple, you just need to push more commits to the same branch -- and the PR will be automatically updated. However doing it will also relaunch all the CI builds and it doesn't seem worth doing it for such a minor change.